BuildScript.Nuke/GenerateSbom/SolutionSbomGenerator.cs

```diff
@@ -0,0 +1,177 @@
+        _packageBlackList,
+        _projectBlacklist,
+        _workingDirectory / cleanedSbomFileName);
+
+    Log.Information("Created cleaned Sbom '{_outputFile}'.", _workingDirectory / cleanedSbomFileName);
+
+    if (_pathToPackageJson == null)
+    {
+      Log.Information("No package json file specified, will therefore not create combined Sbom.");
+      return;
+    }
+
+    Log.Information("Downloading npm Sbom tool...");
+
+    var packageJsonFolder = _pathToPackageJson;
+    if (File.Exists(_pathToPackageJson))
+      packageJsonFolder  = Path.GetDirectoryName(_pathToPackageJson);
+
+    NpmTasks.Npm("install", packageJsonFolder);
+
+    NpmTasks.NpmInstall(_ => _
+        .SetPackages("@cyclonedx/cyclonedx-npm")
+        .SetProcessWorkingDirectory(_workingDirectory));
+
+    var npmSbomFileName = "npm-sbom.xml";
+
+    Log.Information("Creating npm Sbom...");
+
+    PowerShellTasks.PowerShell(_ => _
+        .SetCommand($".\\node_modules\\.bin\\cyclonedx-npm.ps1 {packageJsonFolder /"package.json"} --omit dev --output-format xml --output-file '{npmSbomFileName}'")
+        .SetProcessWorkingDirectory(_workingDirectory));
+
+    Log.Information("Created npm Sbom at '{npmSbomLocation}'.", _workingDirectory / npmSbomFileName);
+
+    var cycloneDXMergeToolPath = _workingDirectory / "cyclonedxtool" / "cyclonedx-cli.exe";
+    Directory.CreateDirectory(cycloneDXMergeToolPath.Parent);
+
+    if (!File.Exists(cycloneDXMergeToolPath))
+    {
+      Log.Information("CycloneDX merge tool not found in working directory. Will download tool from github.");
+      await DownloadCycloneDxMergeTool(cycloneDXMergeToolPath);
+    }
+
+    Log.Information("Merging npm and .net Sboms...");
+    PowerShellTasks.PowerShell(_ => _
+        .SetCommand($"{cycloneDXMergeToolPath} merge --input-files {cleanedSbomFileName} {npmSbomFileName} --output-file {_outputFile}")
+        .SetProcessWorkingDirectory(_workingDirectory));
+
+    Log.Information("Merged Sbom created at {0}", _outputFile);
+  }
+
+  private async Task DownloadCycloneDxMergeTool (AbsolutePath cycloneDXMergeToolPath)
+  {
+    using var httpClient = new HttpClient();
+
+    httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", _githubAccessToken);
+
+    var httpRequestMessage = new HttpRequestMessage();
+    httpRequestMessage.Method = HttpMethod.Get;
+    httpRequestMessage.RequestUri = new Uri("https://api.github.com/repos/CycloneDX/cyclonedx-cli/releases");
+    httpRequestMessage.Headers.Add("User-Agent", "my-app");
+
+    var message = await httpClient.SendAsync(httpRequestMessage);
+    var result = await message.Content.ReadAsStringAsync();
+
+    var jsonData = (JArray)(JsonConvert.DeserializeObject(result)?? throw new InvalidOperationException($"Could not deserialize response from {httpRequestMessage.RequestUri}."));
+    var requiredAsset = jsonData.First?["assets"]?.First(jo => jo["name"]?.Value<string>() == "cyclonedx-win-x86.exe");
```

Comment:
Should we pin the version of cyclonedx-cli? Then we can do https://github.com/CycloneDX/cyclonedx-cli/releases/download/v0.27.1/cyclonedx-win-x86.exe as a single call. Plus, we make sure we don't accidentally download bad stuff. A human would at least pause and think when there's weird stuff on the version. It's something I would feel better about given security concious design. Thoughts?